JS/Lesson26/onur/script.js

```diff
@@ -1,14 +1,68 @@
-/* 
-Elements to use:
-li (.post)
-h2 (.post-title)
-p (.post-body)
-*/
+
+
 
 const URL = 'https://jsonplaceholder.typicode.com/posts';
-function fetchPosts() {
+
+const allPosts = document.getElementById('all-posts');
+
+let selectedPost = null;
+
+async function fetchPosts() {
+
+    try {
+        const response = await fetch(URL);
+        const posts = await response.json();
+        posts.forEach((post) => {
+
+            const listItem = document.createElement('li');
+            listItem.classList.add('post');
+
+            const postTitle = document.createElement('h2');
+            postTitle.classList.add('post-title');
+            postTitle.innerText = post.title;
+
+            const postBody = document.createElement('p');
+            postBody.classList.add('post-body');
+            postBody.innerText = post.body;
+
+            listItem.appendChild(postTitle);
+            listItem.appendChild(postBody);
+
+            const editLink = document.createElement('a');
+            editLink.classList.add('button');
+            editLink.classList.add('edit-button');
+            editLink.textContent = 'Edit';
+
+            listItem.appendChild(editLink);
+
+            editLink.addEventListener('click', editPost);
```

Comment:
I don't think you need an event listener for the edit button. Use <a> element and just add the content of the editPost function before this line: `listItem.appendChild(editLink);` so that the <a> element get's correct href attribute.